### PR TITLE
Improve test and build time by using -i flag

### PR DESCRIPTION
**Pull request by ruflin**
See https://github.com/elastic/beats/issues/4755 for details.

### CODE REVIEW

**Comment by ruflin** [comment]
Curious to see if this also speeds up tests on Jenkins as libbeat packages are only built once for all beats. On Travis it probably doesn't make a difference as it is a new environment each time.

**Comment by ruflin** [comment]
Strangely the Jenkins build took longer them previous builds.

**Comment by ruflin** [comment]
I just reliased this does not apply when running the `testsuite` because there the gotestcover package is used. I will check if I can use the -i flag there too.

File: libbeat/scripts/Makefile [hunk 1]
```diff
@@ -142,15 +143,18 @@ prepare-tests:
 .PHONY: unit-tests
 unit-tests: ## @testing Runs the unit tests with coverage.  Race is not enabled for unit tests because tests run much slower.
 unit-tests: prepare-tests
+	go test -i ${GOPACKAGES}
```
Comment (tsg) [inline libbeat/scripts/Makefile]: From the help of `go test`:

```
	-i
	    Install packages that are dependencies of the test.
	    Do not run the test.
```

And indeed `go test -i` doesn't actually run the tests. However, in master `make unit-tests` seems to work fine, so I'm not quite sure what's going on. But this makes me think adding `-i` to the test commands might not do what we expect.

Comment (ruflin) [inline libbeat/scripts/Makefile]: We actually always run `go test -i` followed by `go test`. Just in the case here it is in the line below. Also see line 151 as an example. My tests showed that running `go test -i` followed by `go test` takes the same time together as just running `go test` alone. So why do it? If you run it more then once, `go test -i` is very quick and the run time will be much shorter.

Comment (tsg) [inline libbeat/scripts/Makefile]: Gotcha, I missed that we run the `go test` without `-i` afterwards. Thanks!

**Comment by cyrilleverrier** [comment]
Will you merge this improvement to the 5.x branches? it is a very nice improvement.

**Comment by ruflin** [comment]
@cyrilleverrier No plan to backport this as it's not directly a feature. I would also give this some more time to be tested. I kind of still worry about side effects as there must be a reason this is not enabled in Golang by default.

**Pull request merged**